community/kernel/src/test/java/org/neo4j/test/ConfigBuilder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.graphdb.config.Setting;
+
+public final class ConfigBuilder
+{
+    public static ConfigBuilder configure( Setting<?> key, String value )
+    {
+        Map<Setting<?>,String> config = new HashMap<>();
+        config.put( key, value );
+        return new ConfigBuilder( config );
+    }
+
+    private final Map<Setting<?>,String> config;
+
+    private ConfigBuilder( Map<Setting<?>,String> config )
+    {
+        this.config = config;
+    }
+
+    public Map<Setting<?>,String> configuration()
+    {
+        return Collections.unmodifiableMap( config );
+    }
+
+    public ConfigBuilder and( Setting<?> key, String value )
```

Comment:
Given that the `configure(...)` method is a static factory method, and the constructor is private, invoking this method will never lead to a `NullPointerException`.